LunrCore/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Lunr;
+
+internal static class DictionaryExtensions
+{
+    public static void Increment(this Dictionary<string, int> dic, string fieldName, int amount = 1)
+    {
+#if NET6_0_OR_GREATER
+
+        ref int value = ref CollectionsMarshal.GetValueRefOrAddDefault(dic, fieldName, out _);
```

Comment:
Please add a comment explaining why this is better (bonus points if you can link to some docs)

Reply:
Using CollectionsMarshal allows us to access the backing field directly (by reference) and avoid an extra lookup and set operation. 

Benchmarking shows a ~2x iimprovement

```
|
BenchmarkDotNet=v0.13.1, OS=debian 11 (container)
Intel Xeon Platinum 8168 CPU 2.70GHz, 1 CPU, 4 logical and 2 physical cores
  [Host] : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT

Toolchain=InProcessEmitToolchain  InvocationCount=1  IterationCount=20  
UnrollFactor=1  WarmupCount=2  

|               Method |     Mean |   Error |  StdDev |
|--------------------- |---------:|--------:|--------:|
| CollectionsMarshalBM | 114.0 μs | 0.32 μs | 0.34 μs |
|         DictionaryBM | 204.6 μs | 2.54 μs | 2.72 μs |
```

Reply:
That's awesome.